### PR TITLE
[Language Server] Support for Formatting

### PR DESCRIPTION
**Pull request by cdietrich**
Implemented Formatting for LanguageServer #21

### CODE REVIEW

**Comment by cdietrich** [comment]
please review

**Comment by svenefftinge** [comment]
Cool! 
I don't like to support two different formatting hooks in future. I think we should deprecate the old one and support only the new one here. @meysholdt what do you think?

**Comment by cdietrich** [comment]
well the old formatter is still the default

**Comment by svenefftinge** [comment]
I think that should be fixed.

**Comment by cdietrich** [comment]
so i will remove the support for the old formatter here and simly return a empty list?

**Comment by meysholdt** [comment]
@svenefftinge yes, it's a good idea to not support the old formatter.
@cdietrich In Xtext languages that use the new generator infrastructure, the new formatter is the default. Only when you use the old generator infrastructure *and* don't have a FormatterFragment in your workflow, then the old formatter is bound in Guice by default. That was for API compatibility purposes, but I don't see any practical use case for it, because the old formatter by default only turned every document into a single line.

**Comment by svenefftinge** [comment]
So we should deprecate the old one and make the new one public API soon, but it is not mandatory to only support the new formatter in LSP. Correct?

**Comment by cdietrich** [comment]
@meysholdt  why is the old formatten then used inside the test language?

**Comment by cdietrich** [comment]
i removed the support for the old formatter now.

File: org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeNewFormatterSetup.xtend [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import org.eclipse.xtext.ide.tests.testlanguage.TestLanguageStandaloneSetup
+import org.eclipse.xtext.ide.tests.testlanguage.formatting2.TestLanguageFormatter
+import org.eclipse.xtext.ide.tests.testlanguage.testLanguage.TestLanguagePackage
+import org.eclipse.xtext.resource.IResourceFactory
+import org.eclipse.xtext.resource.IResourceServiceProvider
+
+/**
+ * Initialization support for running Xtext languages without Equinox extension registry.
+ */
+class TestLanguageIdeNewFormatterSetup extends TestLanguageStandaloneSetup {
```
Comment (meysholdt) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeNewFormatterSetup.xtend]: could you make the formatter impl part of the TestLanguage? Then this special setup should not be necessary anymore.

Comment (cdietrich) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeNewFormatterSetup.xtend]: so i should not test the case having no formatter?

Comment (svenefftinge) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeNewFormatterSetup.xtend]: Yes. testing that is not worth the extra setup.

Comment (cdietrich) [inline org.eclipse.xtext.ide.tests/testlang-src/org/eclipse/xtext/ide/tests/testlanguage/ide/TestLanguageIdeNewFormatterSetup.xtend]: ok ill remove that ;)

**Comment by meysholdt** [comment]
thx. please rebase & squash the commits from this PR, then I think it's ready for being merged. 
see https://github.com/eclipse/xtext/blob/master/QUALITY_GUIDELINES.md#commits-and-pull-requests

**Pull request merged**